wordpress-mobile/WordPress-Android
# WP Stories : Story Title UI/UX fixes

Fixes #12334

Associated PortKey PR https://github.com/Automattic/portkey-android/pull/410
## Findings

## Solution
The main solution here is the Story Title is being persisted as the post title and the Story Title's `EditText` has been optimized for optimal behavior such as requesting focus at the appropriate time and ensuring that the cursor always shows as expected. 

## Testing

1. Create a new story post. 
2. Add a story title. 
3. Close the bottom sheet. 
4. Ensure that you see the story title when you reopen it. 

-------------------

1. Create a new story post. 
2. Add a story title. 
3. Navigate to the tags screen
4. Navigate back to the home screen of the bottom sheet and ensure that the title exists. 

----------------------

1. Create a new story post. 
2. Add a story title. 
3. Close the bottom sheet. 
4. Ensure that you see the story title `EditText` cursor at the end of the content. 

------------------

1. Create a new story post. 
2. Add a story title. 
3. Publish the story. 
4. Ensure that you see the story title you created in the snackbar on the My Site fragment. 

Before | After 
--------|-------
<kbd><img src="https://user-images.githubusercontent.com/1509205/86841922-24853f00-c06a-11ea-8506-e08119d9c55f.png" width="320"></kbd> |       <kbd><img src="https://user-images.githubusercontent.com/1509205/86841939-26e79900-c06a-11ea-9797-edd811d61952.png" width="320"></kbd>

------------------

1. Create a new story post. 
2. Do not add a story title.
3. Publish the story. 
4. Ensure that you see the "Untitled" in the snackbar on the My Site fragment. 

Before | After 
--------|-------
<kbd><img src="https://user-images.githubusercontent.com/1509205/86845130-a8412a80-c06e-11ea-9443-6ade146f386e.png" width="320"></kbd> |       <kbd><img src="https://user-images.githubusercontent.com/1509205/86845156-b55e1980-c06e-11ea-8b7d-ca0f668a46bd.png" width="320"></kbd>

## Concerns
So some time was spent while working on this PR to fix the keyboard focus issues but an optimal solution was not found so the associated issue will remain open with the items that aren't checked so they can be addressed later.

## Reviewing 
Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist
